PyU4V/tools/openstack/migrate_utils.py

```diff
@@ -0,0 +1,866 @@
+            txt_out = self.input(txt)
+            if self.check_input(txt_out, 'Y'):
+                volume_list, create_vol = self.choose_subset_volumes(
+                    storage_group_name, volume_list)
+            else:
+                sys.exit()
+        return volume_list, create_vol
+
+    def get_volume_list(self, storage_group_name):
+        """Get the list of volumes from the storage group.
+
+        :param storage_group_name: the storage group name
+        :returns: List, Boolean
+        """
+        create_vol = False
+        volume_list = self.conn.provisioning.get_volumes_from_storage_group(
+            storage_group_name)
+        print_str = 'There are %d volume in storage group %s.'
+        self.smart_print(print_str, DEBUG, len(volume_list),
+                         storage_group_name)
+        txt = ('Do you want to migrate all %s volumes: Y/N: '
+               % len(volume_list))
+        txt_out = self.input(txt)
+        if self.check_input(txt_out, 'Y'):
+            # Move list of devices from old to new masking view
+            print_str = ('Moving all volumes between source '
+                         'and target storage groups.')
+            self.smart_print(print_str, DEBUG)
+            create_vol = True
+        else:
+            volume_list, create_vol = self.choose_subset_volumes(
+                storage_group_name, volume_list)
+
+        return volume_list, create_vol
+
+    def choose_sg(
+            self, masking_view, child_sg_list, portgroup, initiatorgroup,
+            revert=False):
+        """Choose a child storage group from the list.
+
+        :param masking_view: the masking view
+        :param child_sg_list: the storage group list
+        :param portgroup: The port group name
+        :param initiatorgoup: the initiator group name
+        :param revert: revert back
+        :returns: dict, string
+        """
+        element_dict = {}
+        child_sg = None
+        for child_sg in child_sg_list:
+            txt = ('Which storage group do you want to migrate:\n\t '
+                   '%s. Y/N: ' % child_sg)
+            txt_out = self.input(txt)
+            if 'Y' in txt_out:
+                # Compile the new names of the SGs and MV
+                element_dict = self.compile_new_element_names(
+                    masking_view, portgroup, initiatorgroup, child_sg,
+                    revert)
+                return element_dict, child_sg
+        return element_dict, child_sg
+
+    def validate_masking_view(self, masking_view, revert=False):
+        """Validate the masking view string.
+
+        :param masking_view: masking view name
+        :param revert: revert back
+        :returns: boolean
+        """
+        if re.search('^OS-', masking_view):
+            try:
+                if revert:
+                    return True if self.get_mv_component_dict(
+                        masking_view, sys.argv[1]) else False
```

Comment:
sys.argv[1] may need cast as a boolean here, get_mv_component_dict() expects boolean value for revert input param